### DISCUSSION TITLE
Where to register vue component?

### DISCUSSION THREAD

**Discussion by sander-vth** [statamic/cms#6865] [Q&A]
Hi,

I am new to Statamic.
I would like to register a Vue component.
But I'm unsure on where and how?

I have 

`import MyComponent from './components/Listing';

Statamic.booting(() => {
    Statamic.$components.register('sample-listing', MyComponent );
});
`

I have put this in public/vendor/app/js/cp.js.
And my vue file is in public/vendor/app/js/components/Listing.vue
This file gets loaded, but I'm unable to use [sample-listing]  in my view files.

What step(s) am I missing?

**Comment by joshkrz** [comment]
I'm experiencing the same issue, the Vue component is showing up in dev tools but it will not render out on the page. I think it is because we are using Vue 3 elsewhere so the component is being compiled down to 3 when the CP uses 2. Still figuring out how to using Vue 2 and 3 at the same time in Vite. No luck so far.
